DSESwift/DataSyncer.swift

```diff
@@ -8,37 +8,398 @@
+            // remove the deletion request which has been processed
+            try DatabaseHandler.deleteDataDeletionRequest(request.uuid)
+        }
+    }
+    
+    
+    func downloadSensorsFromRemote() throws {
+        do{
+            let sensors = try getSensorsFromRemote()
+            try insertSensorsIntoLocal(sensors)
+            //notify
+            self.delegate?.onSensorsDownloadCompleted()
+        }catch {
+            self.delegate?.onSensorDataDownloadFailed(error)
+            throw error
+        }
+
+    }
+    
+    func downloadSensorsDataFromRemote() throws {
+        do{
+            let sensorsInLocal = getSensorsInLocal()
+            for sensor in sensorsInLocal{
+                if (sensor.remoteDownloadEnabled){
+                    try downloadAndStoreDataForSensor(sensor)
+                    try updateDownloadStatusForSensor(sensor)
+                }
+            }
+            self.delegate?.onSensorDataDownloadCompleted()
+        }catch{
+            self.delegate?.onSensorDataDownloadFailed(error)
+            throw error
+        }
+    }
+
+    func uploadSensorDataToRemote() throws {
+        let sensorsInLocal = getSensorsInLocal()
+        for sensor in sensorsInLocal {
+            print("###", sensor.name)
+            print("###", sensor.remoteUploadEnabled)
+            if (sensor.remoteUploadEnabled){
+                let unuploadedDataPoints = try getUnuploadedDataPoints(sensor)
+                print("### datapoints", unuploadedDataPoints)
+                try uploadDataPointsForSensor(unuploadedDataPoints, sensor: sensor)
+                try updateUploadStatusForDataPoints(unuploadedDataPoints)
+            }
+        }
+    }
+    
+    func cleanLocalStorage() throws {
+        let sensorsInLocal = getSensorsInLocal()
+        for sensor in sensorsInLocal {
+            try purgeDataForSensor(sensor)
+        }
+    }
+    
+    // MARK: Helper functions
+    private func downloadAndStoreDataForSensor(sensor: Sensor) throws{
+        let limit = 1000
+        var isCompleted = false
+        let persistentBoundary = NSDate().dateByAddingTimeInterval (-1 * persistentPeriod)
+        var startBoundaryForNextQuery: NSDate?
+        
+        // download and store data. Loop stops when the number of datapoints is not equal to the limit specified in the request.
+        repeat{
+            var queryOptions = QueryOptions()
+            queryOptions.limit = limit
+            queryOptions.startTime = (startBoundaryForNextQuery == nil) ? persistentBoundary : startBoundaryForNextQuery
+            let sensorData = try SensorDataProxy.getSensorData(sourceName: sensor.source, sensorName: sensor.name)
+            try insertSensorDataIntoLocalDB(sensorData, sensorId: sensor.id)
+            
+            //check if download is completed, if not prepare for the next download
+            isCompleted = isDataDownloadCompleted(sensorData, limit: limit)
```

Comment:
this method is only used once and it is a oneliner. I would argue to put it in here and actually explain why this check is good enough.